src/Hedgehog.Experimental/TypeShape.fs

```diff
@@ -0,0 +1,20 @@
+[<AutoOpen>]
+module Hedgehog.TypeShape
+
+open System
+open TypeShape.Core
+
+type GenericArgument = {
+  argType: Type
+  argTypeDefinition: Type
+}
+
+let (|GenericShape|_|) (s: TypeShape) =
```

Comment:
To every public function, please add documentation and an explicit return type

Reply:
Oops, didn't mean it to be public. Will make it `internal` instead.

Reply:
Made the module `internal`, done.